packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_screen.dart

```diff
@@ -24,8 +130,112 @@ class DeepLinksScreen extends Screen {
+    return const DeepLinkPage();
+  }
+}
+
+class DeepLinkPage extends StatefulWidget {
+  const DeepLinkPage({super.key});
+
+  @override
+  State<DeepLinkPage> createState() => _DeepLinkPageState();
+}
+
+class _DeepLinkPageState extends State<DeepLinkPage> {
+  String responseBody = 'empty response';
+  int? selectedRowIndex;
+  String searchContent = '';
+  bool bundleByDomain = false;
+
+  void setBundleByDomain(bool shouldBundleByDomain) {
+    setState(() {
+      bundleByDomain = shouldBundleByDomain;
+    });
+  }
+
+  void setResponse(String response) {
+    setState(() {
+      responseBody = response;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView(
+      children: [
+        Row(
+          children: [
+            Expanded(child: buildTitle('All deep links', context)),
+            InkWell(
+              onTap: () {
+                setBundleByDomain(!bundleByDomain);
+              },
+              child: const Text('Bundle by domain'),
+            ),
+            const SizedBox(width: 10),
+            SearchBar(
+              leading: const Icon(Icons.search),
+              onChanged: (value) {
+                setState(() {
+                  searchContent = value.toLowerCase();
+                });
+              },
+              constraints: BoxConstraints.tight(const Size(200, 40)),
+            ),
+          ],
+        ),
+        const SizedBox(height: 10),
+        buildDataTable(context),
+      ],
     );
   }
+
+  Widget buildDataTable(BuildContext context) {
+    List<LinkData> linkDatas = searchContent.isNotEmpty
+        ? allLinkDatas
+            .where((linkData) => linkData.searchLabel.contains(searchContent))
+            .toList()
+        : allLinkDatas;
+
+    if (bundleByDomain) {
+      final Map<String, LinkData> bundleByDomainMap = {};
+      for (var linkData in linkDatas) {
+        bundleByDomainMap[linkData.domain] =
+            linkData.mergeByDomain(bundleByDomainMap[linkData.domain]);
+      }
+      linkDatas = bundleByDomainMap.values.toList();
+    }
+
+    final List<DataRow> rows = [
+      for (var i = 0; i < linkDatas.length; i++)
+        linkDatas[i].buildRow(
+          context,
+          color: i % 2 == 0
+              ? MaterialStateProperty.all<Color>(
+                  Theme.of(context).colorScheme.onSurface.withOpacity(0.2),
+                )
+              : null,
+        ),
+    ];
+
+    return DataTable(
+      columns: const [
+        DataColumn(label: Text('OS')),
+        DataColumn(label: Text('Scheme')),
+        DataColumn(label: Text('Domain')),
+        DataColumn(label: Text('Path')),
+      ],
+      dataRowMinHeight: bundleByDomain ? 150 : null,
+      dataRowMaxHeight: bundleByDomain ? 150 : null,
+      rows: rows,
+    );
+  }
+}
+
+Widget buildTitle(String text, BuildContext context) {
+  final textTheme = Theme.of(context).textTheme;
+  return Text(
```

Comment:
declare this Text widget directly in the build method it is used in above instead of having a method that returns a widget. This is too simple to warrant creating its own stateless widget class